src/Grid/grid.jl

```diff
@@ -73,37 +62,176 @@ the vertices that define an *oriented face*. This function induces the
+    )
+end
+function faces(c::AbstractCell{3, RefPrism})
+    ns = get_node_ids(c)
+    return (
+        (ns[1], ns[3], ns[2]),        (ns[1], ns[2], ns[5], ns[4]), # f1, f2
+        (ns[3], ns[1], ns[4], ns[6]), (ns[2], ns[3], ns[6], ns[5]), # f3, f4
+        (ns[4], ns[5], ns[6]),                                      # f5
+    )
+end
+
+
+######################################################
+# Concrete implementations of AbstractCell interface #
+######################################################
+
+# Lagrange interpolation based cells
+struct Line                   <: AbstractCell{1, RefInterval}      nodes::NTuple{ 2, Int} end
+struct QuadraticLine          <: AbstractCell{1, RefInterval}      nodes::NTuple{ 3, Int} end
+struct Triangle               <: AbstractCell{2, RefTriangle}      nodes::NTuple{ 3, Int} end
+struct QuadraticTriangle      <: AbstractCell{2, RefTriangle}      nodes::NTuple{ 6, Int} end
+struct Quadrilateral          <: AbstractCell{2, RefQuadrilateral} nodes::NTuple{ 4, Int} end
+struct QuadraticQuadrilateral <: AbstractCell{2, RefQuadrilateral} nodes::NTuple{ 9, Int} end
+struct Tetrahedron            <: AbstractCell{3, RefTetrahedron}   nodes::NTuple{ 4, Int} end
+struct QuadraticTetrahedron   <: AbstractCell{3, RefTetrahedron}   nodes::NTuple{10, Int} end
+struct Hexahedron             <: AbstractCell{3, RefHexahedron}    nodes::NTuple{ 8, Int} end
+struct QuadraticHexahedron    <: AbstractCell{3, RefHexahedron}    nodes::NTuple{27, Int} end
+struct Wedge                  <: AbstractCell{3, RefPrism}         nodes::NTuple{ 6, Int} end
+
+default_interpolation(::Type{Line})                   = Lagrange{1, RefCube,        1}()
+default_interpolation(::Type{QuadraticLine})          = Lagrange{1, RefCube,        2}()
+default_interpolation(::Type{Triangle})               = Lagrange{2, RefTetrahedron, 1}()
+default_interpolation(::Type{QuadraticTriangle})      = Lagrange{2, RefTetrahedron, 2}()
+default_interpolation(::Type{Quadrilateral})          = Lagrange{2, RefCube,        1}()
+default_interpolation(::Type{QuadraticQuadrilateral}) = Lagrange{2, RefCube,        2}()
+default_interpolation(::Type{Tetrahedron})            = Lagrange{3, RefTetrahedron, 1}()
+default_interpolation(::Type{QuadraticTetrahedron})   = Lagrange{3, RefTetrahedron, 2}()
+default_interpolation(::Type{Hexahedron})             = Lagrange{3, RefCube,        1}()
+default_interpolation(::Type{QuadraticHexahedron})    = Lagrange{3, RefCube,        2}()
+default_interpolation(::Type{Wedge})                  = Lagrange{3, RefPrism,       1}()
+
+# TODO: Deprecate
+const Line2D = Line
+const Line3D = Line
+const Quadrilateral3D = Quadrilateral
+edges(c::Quadrilateral3D) = faces(c)
+
+# Serendipity interpolation based cells
+# TODO: Naming?
+struct SQuadraticQuadrilateral <: AbstractCell{2, RefQuadrilateral} nodes::NTuple{ 8, Int} end
+struct SQuadraticHexahedron    <: AbstractCell{3, RefHexahedron}    nodes::NTuple{20, Int} end
```

Comment:
I think this is a good name, as long as we add some docs to clarify the naming convention.

Reply:
What about using just `Q` for quadratic too then? `QTriangle`, `QQuadrilateral`, `SQQuadrilateral` etc.?

Reply:
I'm more skeptical about that, to me that's a bit hard to interpret. 
I prefer the current version being spelled out (it is not so often one has to spell it, so avoiding e.g. `SerendipityQuadraticQuadrilateral` is IMO mostly to avoid having too long types when printing e.g. error messages and stack-traces.

Reply:
Since we are discussing naming: Another problem for higher order elements which we might want to consider is that we do not really give information about the node placement (currently only equidistant, but could also be e.g. GLL placement).